### PR TITLE
cnap/core: add the implementation for pipeline related classes

### LINKED ISSUE #52: design pipeline related classes to manage the pipeline

### PR DESCRIPTION
**Pull request by Hulongyin**
This PR is to fix #52 , design and implement for pipeline related classes.

### CODE REVIEW

File: cnap/core/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        Return: None
+        """
+        LOG.debug("Register Pipeline: %s", str(dict(pipeline_obj)))
+        try:
+            self._db.save_table_object_dict(
+                PipelineManager.PIPELINE_TABLE,
+                pipeline_obj.id,
+                dict(pipeline_obj)
+                )
+        except ValueError as e:
```
Comment (leyao-daily) [inline cnap/core/pipeline.py]: After check with the python, I think we are catching a ValueError and re-raising the same error. We can just simplify it with
```
LOG.debug("Register Pipeline: %s", str(dict(pipeline_obj)))
        self._db.save_table_object_dict(
            PipelineManager.PIPELINE_TABLE,
            pipeline_obj.id,
            dict(pipeline_obj)
        )
```
And now, any ValueError exceptions will naturally propagate up to the caller in Python machinism.

Comment (Hulongyin) [reply cnap/core/pipeline.py]: Thanks, fixed.

File: cnap/core/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+"""
+This module contains the definition of the `Pipeline` class and `PipelineManager` class.
+The `Pipeline` class contains the pipeline information including pipeline id, stream provdier
```
Comment (hwang37) [inline cnap/core/pipeline.py]: A typo stream provdier => stream provider

Comment (Hulongyin) [reply cnap/core/pipeline.py]: Thanks, fixed.

**Review by leyao-daily — APPROVED** [review]

**Comment by Hulongyin** [comment]
Closes #52

**Pull request merged**